components/roode/ranging.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include "VL53L1X_ULD.h"
+
+namespace esphome {
+namespace roode {
+
+struct RangingConfig {
+    explicit RangingConfig(uint16_t timing_budget) : mode{timing_budget == 15 ? Short : Long},
+                                                     timing_budget{timing_budget} {}
+
+    explicit RangingConfig(uint16_t timing_budget, EDistanceMode mode) : mode{mode},
+                                                                         timing_budget{timing_budget} {}
+
+    EDistanceMode const mode;
+    uint16_t const timing_budget;
+    uint16_t const delay_between_measurements = timing_budget + 5;
+};
+
+namespace Ranging {
+    static const RangingConfig *Shortest = new RangingConfig(15);
+    static const RangingConfig *Short = new RangingConfig(20);
+    static const RangingConfig *Medium = new RangingConfig(33);
+    static const RangingConfig *MediumLong = new RangingConfig(50);
+    static const RangingConfig *Long = new RangingConfig(100);
+    static const RangingConfig *Longer = new RangingConfig(200);
+    static const RangingConfig *Longest = new RangingConfig(500);
+
+    static const RangingConfig *Custom(uint16_t timing_budget) { return new RangingConfig(timing_budget); };
```

Comment:
`timing_budget` must be one of the above defined budgets, because you can't set a custom budget with the ULD API. That's only possible with STs full API.

Reply:
Yes I actually saw that as well in the library API...but then promptly forgot when refactoring the sensor mode calibration. That's great too because that Custom function was giving some compiler warnings and I couldn't figure out why.

I was trying to expose more raw library calibration configuration parameters, which is why I have
```yaml
  calibration:
    # both these next 2 default to "auto" for formerly "calibration: true"
    distance_mode: long/short
    timing_budget: 33/100/200...
```
But looking back I think they should just be consolidated. `distance_mode` can be inferred since it's only `Short` for `15`.
Maybe we still expose it as `distance_mode` but have it 1-1 to `RangingConfig` presets. With that in mind it makes less since to make the raw numbers available since those are for the timing budget.